panic: interface conversion in processJobEvent

### Problem description
Non descriptive/unhelpful error message when creation of a job resource fails.

The `doesnt_exist.sh` script in the example below does not exist in the image, so creation of the job fails. Kubernetes will retry creating the resource 5 times and then fail, causing pulumi to crash with the `runtime.Object is nil, not *unstructured.Unstructured` which isn't very helpful. Especially since there's tons of things that might cause the creation of a Job to fail, but this error will cause the whole deployment to stop with `transport is closing` errors, making it very hard to figure out which resource is causing the issue.

### Errors & Logs


```
panic: interface conversion: runtime.Object is nil, not *unstructured.Unstructured
    goroutine 178 [running]:
    github.com/pulumi/pulumi-kubernetes/pkg/await.(*jobInitAwaiter).processJobEvent(0xc000a87950, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/pulumi/pulumi-kubernetes/pkg/await/job.go:183 +0x77d
    github.com/pulumi/pulumi-kubernetes/pkg/await.(*jobInitAwaiter).Await(0xc000a87950, 0x0, 0x0)
        /home/travis/gopath/src/github.com/pulumi/pulumi-kubernetes/pkg/await/job.go:134 +0x4dd
    github.com/pulumi/pulumi-kubernetes/pkg/await.glob..func4(0xc0003d0000, 0x1a96f40, 0xc0002b34c0, 0xc000e16a80, 0x51, 0xc000ddb2d8, 0x8, 0xc0010584c0, 0xc0004e3470, 0xc000011e50, ...)
        /home/travis/gopath/src/github.com/pulumi/pulumi-kubernetes/pkg/await/awaiters.go:155 +0x4b
    github.com/pulumi/pulumi-kubernetes/pkg/await.Creation(0x1a96f40, 0xc0002b34c0, 0xc0003d0000, 0xc000e16a80, 0x51, 0xc000ddb2d8, 0x8, 0xc0004e3470, 0xc0010584c0, 0x1a5a5a0, ...)
        /home/travis/gopath/src/github.com/pulumi/pulumi-kubernetes/pkg/await/await.go:190 +0x661
    github.com/pulumi/pulumi-kubernetes/pkg/provider.(*kubeProvider).Create(0xc0002c2210, 0x1a97000, 0xc000ec4870, 0xc000a93740, 0xc0002c2210, 0x1635b01, 0xc000b19140)
        /home/travis/gopath/src/github.com/pulumi/pulumi-kubernetes/pkg/provider/provider.go:1152 +0x65f
    github.com/pulumi/pulumi/sdk/proto/go._ResourceProvider_Create_Handler.func1(0x1a97000, 0xc000ec4870, 0x1756fa0, 0xc000a93740, 0x1782860, 0x26b2a08, 0x1a97000, 0xc000ec4870)
        /home/travis/gopath/pkg/mod/github.com/pulumi/pulumi@v1.5.2-0.20191119200129-f9085bf79966/sdk/proto/go/provider.pb.go:1615 +0x86
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1(0x1a97000, 0xc000eb79e0, 0x1756fa0, 0xc000a93740, 0xc0010bf980, 0xc0010bf9a0, 0x0, 0x0, 0x1a57fa0, 0xc000283840)
        /home/travis/gopath/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20171105060200-01f8541d5372/go/otgrpc/server.go:61 +0x36e
    github.com/pulumi/pulumi/sdk/proto/go._ResourceProvider_Create_Handler(0x17c2bc0, 0xc0002c2210, 0x1a97000, 0xc000eb79e0, 0xc000ddf900, 0xc0003d0080, 0x1a97000, 0xc000eb79e0, 0xc000eba000, 0x3bb)
        /home/travis/gopath/pkg/mod/github.com/pulumi/pulumi@v1.5.2-0.20191119200129-f9085bf79966/sdk/proto/go/provider.pb.go:1617 +0x14b
    google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004a0d80, 0x1ab2ca0, 0xc000492600, 0xc000e15400, 0xc0002921b0, 0x267e390, 0x0, 0x0, 0x0)
        /home/travis/gopath/pkg/mod/google.golang.org/grpc@v1.21.1/server.go:998 +0x46a
    google.golang.org/grpc.(*Server).handleStream(0xc0004a0d80, 0x1ab2ca0, 0xc000492600, 0xc000e15400, 0x0)
        /home/travis/gopath/pkg/mod/google.golang.org/grpc@v1.21.1/server.go:1278 +0xd97
    google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0004de030, 0xc0004a0d80, 0x1ab2ca0, 0xc000492600, 0xc000e15400)
        /home/travis/gopath/pkg/mod/google.golang.org/grpc@v1.21.1/server.go:717 +0xbb
    created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/travis/gopath/pkg/mod/google.golang.org/grpc@v1.21.1/server.go:715 +0xa1

    error: update failed
```

### Versions
$ pulumi version
v1.7.1

$ pulumi plugin ls
NAME        KIND      VERSION  SIZE   INSTALLED  LAST USED
gcp         resource  2.0.0    78 MB  n/a        21 hours ago
kubernetes  resource  1.4.0    49 MB  n/a        21 hours ago
random      resource  1.3.0    36 MB  n/a        1 day ago

### Reproducing the issue
```
    import pulumi

    from pulumi_kubernetes.batch.v1 import Job

    config = pulumi.Config()
    stack = pulumi.get_stack()
    project = pulumi.get_project()

    name = '{}-post-install-postgres'.format(stack)

    return Job(
        name,
        metadata={
            'labels':
                {
                    'app': project,
                    'release': stack
                }
        },
        spec={
            'template':{
                'metadata': {
                    'name': name,
                    'labels':
                        {
                            'app': project,
                            'release': stack
                        }
                },
                'spec':{
                    'restartPolicy':'Never',
                    'containers':[{
                        'name': 'init-databases',
                        'image': 'postgres:9.6',
                        'command': ["/bin/bash", "doesnt_exist.sh"],
                    }]
                }
            }
        },
        opts=pulumi.ResourceOptions(depends_on=dependencies)
    )
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 ## HEAD (Unreleased)
 
+### Bug fixes
+
+-   Fix deprecation warnings and docs. (https://github.com/pulumi/pulumi-kubernetes/pull/929).
+
 ## 1.4.1 (December 17, 2019)
 
 ### Bug fixes

--- a/pkg/await/job.go
+++ b/pkg/await/job.go
@@ -180,12 +180,11 @@ func (jia *jobInitAwaiter) Read() error {
 }
 
 func (jia *jobInitAwaiter) processJobEvent(event watch.Event) error {
-	uns, ok := event.Object.(*unstructured.Unstructured)
-	if !ok {
+	if event.Object == nil {
 		glog.V(3).Infof("received event with nil Object: %#v", event)
 		return nil
 	}
-	job, err := clients.JobFromUnstructured(uns)
+	job, err := clients.JobFromUnstructured(event.Object.(*unstructured.Unstructured))
 	if err != nil {
 		glog.V(3).Infof("Failed to unmarshal Job event: %v", err)
 		return nil

--- a/pkg/await/job.go
+++ b/pkg/await/job.go
@@ -180,7 +180,12 @@ func (jia *jobInitAwaiter) Read() error {
 }
 
 func (jia *jobInitAwaiter) processJobEvent(event watch.Event) error {
-	job, err := clients.JobFromUnstructured(event.Object.(*unstructured.Unstructured))
+	uns, ok := event.Object.(*unstructured.Unstructured)
+	if !ok {
+		glog.V(3).Infof("received event with nil Object: %#v", event)
+		return nil
+	}
+	job, err := clients.JobFromUnstructured(uns)
 	if err != nil {
 		glog.V(3).Infof("Failed to unmarshal Job event: %v", err)
 		return nil

